src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerSearchViewModel.cs

```diff
@@ -462,7 +462,19 @@ private void PackageOnExecuted(PackageManagerSearchElement element, PackageVersi
                     return;
                 }
 
-                var allPackageVersions = PackageManagerSearchElement.ListRequiredPackageVersions(headers, version);
+                List<Tuple<PackageHeader, PackageVersion>> allPackageVersions = null;
+                try
+                {
+                    allPackageVersions = PackageManagerSearchElement.ListRequiredPackageVersions(headers, version).ToList();
+                }
+                catch
+                {
+                    MessageBox.Show(String.Format("Could not find all dependencies for package {0}", element.Name),
+                    Resources.PackageDownloadErrorMessageBoxTitle,
+                    MessageBoxButton.OK, MessageBoxImage.Error);
+
+                    return;
+                }
 
```

Comment:
maybe we should check `allPackageVersions == null` adn post the err nessage outside the `catch`....to be sure no nulls get passed on..

Reply:
Can `ToList()` return `null` instead of throwing?

Reply:
`ToList`
https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.tolist?view=netframework-4.7.2

Exceptions
ArgumentNullException
source is null.

Reply:
I guess it throws if the argument is null...